src/main/java/com/teamabnormals/abnormals_core/core/AbnormalsCore.java

```diff
@@ -84,12 +86,17 @@ public AbnormalsCore() {
 			}
 		});
 
-		DistExecutor.runWhenOn(Dist.CLIENT, () -> () -> {
-			((IReloadableResourceManager) Minecraft.getInstance().getResourceManager()).addReloadListener(ENDIMATION_DATA_MANAGER);
-			modEventBus.addListener(this::clientSetup);
-		});
+		DistExecutor.unsafeRunWhenOn(Dist.CLIENT, () -> () -> modEventBus.addListener(EventPriority.NORMAL, false, ColorHandlerEvent.Block.class, event ->
```

Comment:
What's the reasoning behind the event being `ColorHandlerEvent.Block`?

Reply:
It's based on JEI [here.](https://github.com/mezz/JustEnoughItems/blob/1.16/src/main/java/mezz/jei/JustEnoughItems.java#L33-L52)

Reply:
This is based on JEI [here](https://github.com/mezz/JustEnoughItems/blob/1.16/src/main/java/mezz/jei/JustEnoughItems.java#L33-L52). A texture atlas is added to the resource manager without issue.

Reply:
That doesn't quite answer my question.
The reasoning that you're stating here as to why it's listening to `ColorHandlerEvent.Block` is that it works?
What I was asking is _why_ it works, not _how_ it works. 
I'm only asking this because I want to make sure this is the best way of doing it.

Reply:
`ColorHandlerEvent.Block` is the first event fired after the resource manager is initialized, but only if Minecraft has been initialized. Using data runs does not initialize Minecraft so the instance is null. This verifies the listener can be added if the game is running normally.